packages/amplify_auth_cognito/ios/Classes/AuthCognitoDefault.swift

```diff
@@ -0,0 +1,174 @@
+        _ = Amplify.Auth.update(oldPassword: request.oldPassword, to: request.newPassword) { response in
+            switch response {
+            case .success:
+                let emptyMap: Dictionary<String, Any> = [:]
+                flutterResult(emptyMap)
+                
+            case .failure(let updatePasswordError):
+                handleAuthError(error: updatePasswordError, flutterResult: flutterResult, msg: FlutterAuthErrorMessage.UPDATE_PASSWORD.rawValue)
+            }
+        }
+    }
+    
+    func onResetPassword(flutterResult: @escaping FlutterResult, request: FlutterResetPasswordRequest) {
+        _ = Amplify.Auth.resetPassword(for: request.username) { response in
+            switch response {
+            case .success:
+                let resetData = FlutterResetPasswordResult(res: response)
+                flutterResult(resetData.toJSON())
+                
+            case .failure(let resetPasswordError):
+                handleAuthError(error: resetPasswordError, flutterResult: flutterResult,  msg: FlutterAuthErrorMessage.RESET_PASSWORD.rawValue)
+            }
+        }
+    }
+    
+    func onConfirmPassword(flutterResult: @escaping FlutterResult, request: FlutterConfirmPasswordRequest) {
+        _ = Amplify.Auth.confirmResetPassword(for: request.username, with: request.newPassword, confirmationCode: request.confirmationCode) { response in
+            switch response {
+            case .success:
+                let emptyMap: Dictionary<String, Any> = [:]
+                flutterResult(emptyMap)
+                
+            case .failure(let resetPasswordError):
+                handleAuthError(error: resetPasswordError, flutterResult: flutterResult,  msg: FlutterAuthErrorMessage.RESET_PASSWORD.rawValue)
+            }
+        }
+    }
+    
+    func onFetchSession(flutterResult: @escaping FlutterResult, request: FlutterFetchSessionRequest) {
+        _ = Amplify.Auth.fetchAuthSession { result in
+            do {
+                if (request.getAWSCredentials) {
+                    let sessionData = try FlutterFetchCognitoSessionResult(res: result)
+                    flutterResult(sessionData.toJSON())
+                } else {
+                    let sessionData = try FlutterFetchSessionResult(res: result)
+                    if (sessionData.isSignedIn) {
+                        flutterResult(sessionData.toJSON())
+                    } else {
+                        throw AuthError.signedOut("There is no user signed in to retreive user sub",
+                                                  "Call Auth.signIn to sign in a user and then call Auth.fetchSession")
+                    }
+                }
+                
+            } catch {
+                handleAuthError(error: error as! AuthError, flutterResult: flutterResult,  msg: FlutterAuthErrorMessage.FETCH_SESSION.rawValue)
+            }
+        }
+    }
```

Comment:
By following this unit testing strategy, I believe all of this file goes untested. Are we comfortable with this since there is some logic in this file?

Reply:
Personally I think so - the logic is minimal for most of the methods other than `onFetchSession` and `onGetCurrentUser` (mostly just switch cases for .success and .failure).